### PR TITLE
Migrate to codecov.io

### PR DESCRIPTION
**Pull request by junczhuMSFT**
Fix:  #441 

Codecov Migrate Steps:

- [x] add coverage percent at .codecov
- [x] add codecov-action at build.yml
- [x] add badge at README.md

- [x] update make test with go test at makefile
- [x] update .gitignore / rmv .cover/ .test/ add coverage.txt

- [x] del shell/test.sh
- [x] del origin upload coverage report from build.yml

### CODE REVIEW

**Comment by sajayantony** [comment]
@junczhuMSFT might have to rebase this. 
/cc @shizhMSFT @qweeah

**Comment by shizhMSFT** [comment]
@junczhuMSFT You may also want to clean up previous coverage related scripts.

File: Makefile [hunk 1]
```diff
@@ -19,7 +19,7 @@ LDFLAGS += -X $(PROJECT_PKG)/internal/version.GitTreeState=${GIT_DIRTY}
 
 .PHONY: test
 test: vendor check-encoding
-	./scripts/test.sh
+		go test -race -v -coverprofile=coverage.txt -covermode=atomic ./...
```
Comment (shizhMSFT) [inline Makefile:22]: We don't the converge file, do we?

Comment (shizhMSFT) [inline Makefile:22]: There's extra space in the code.

Comment (junczhuMSFT) [reply Makefile]: > There's extra space in the code.

Extra space has been removed

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**